docs/src/code-style.rst

```diff
@@ -2,20 +2,19 @@
 Code Style Guidelines
 *********************
 
+.. _ruff:
 
-.. _black-code-style:
-
-Black
------
+Ruff
+------
 
-We use the `black <https://github.com/psf/black>`_ coding style, a flavour of `PEP-8 <https://www.python.org/dev/peps/pep-0008/>`_ for Python.
+We use `ruff <https://github.com/astral-sh/ruff/>`__ to format and lint our python files.
 
-We use a `pre-commit-hook <https://pre-commit.com/>`_ to apply this style before committing, so you don't have to bother about formatting.
+We use a `pre-commit-hook <https://pre-commit.com/>`_ to apply these rules before committing, so you don't have to bother about doing so manually.
```

Comment:
Suggest `pre-commit run ruff --all-files`?

Reply:
`./tools/ruff.sh` is our wrapper. Since it's mentioned right below this, I'd say let's keep it as-is, especially since the pre-commit hook in turn calls `./tools/ruff.sh`.

Reply:
I'm not sure I understand, these are the docs for this repo, right? 😅 

I almost feel like we should replace this page of the docs with instructions for setting up vscode/nvim/pycharm with appropriate plugins. In theory noone should need to run these tools manually at all :D

Reply:
My point was more about why make this repo work differently than other repos that use pre-commit but it is not important.  It is better to merge these changes.

I agree that most IDEs usually have these tools readily available.

Reply:
Ah, gotcha. Fair. 

> It is better to merge these changes.

Agreed - I think a docs overhaul is a topic for another time 😄